taskcluster/ac_taskgraph/transforms/ui_tests.py

```diff
@@ -0,0 +1,52 @@
+from taskgraph.transforms.base import TransformSequence
+
+
+transforms = TransformSequence()
+
+
+_ANDROID_TASK_NAME_PREFIX = "android-"
+
+
+@transforms.add
+def set_component_attribute(config, tasks):
+    for task in tasks:
+        component_name = task.pop("component", None)
+        if not component_name:
+            task_name = task["name"]
+            if task_name.startswith(_ANDROID_TASK_NAME_PREFIX):
+                component_name = task_name[len(_ANDROID_TASK_NAME_PREFIX):]
+            else:
+                raise NotImplementedError(f"Cannot determine component name from task {task_name}")
+
+        attributes = task.setdefault("attributes", {})
+        attributes["component"] = component_name
+
+        yield task
+
+
+@transforms.add
+def define_ui_test_command_line(config, tasks):
+    for task in tasks:
+        run = task.setdefault("run", {})
+        post_gradlew = run.setdefault("post-gradlew", [])
+        post_gradlew.append(
+            ['automation/taskcluster/androidTest/ui-test.sh', task["attributes"]["component"], 'arm', '1']
+        )
+
+        yield task
+
+
+@transforms.add
+def define_treeherder_symbol(config, tasks):
+    for task in tasks:
+        treeherder = task.setdefault("treeherder")
+        treeherder.setdefault("symbol", f"{task['attributes']['component']}(unit)")
```

Comment:
Great question! In this case, the component (e.g.: `samples-glean`) is the group symbol. `unit` is the job name. The idea is to regroup the test next to the build:

<img width="230" alt="image" src="https://user-images.githubusercontent.com/5907366/205918096-ed17baed-8dbb-4eb4-84f4-e8ac63b8c83b.png">

I took this liberty because I get the feeling it's better to bring related things closer together. That said, I'm happy to get back to the existing behavior 🙂